src/cmap/auth/gssapi.ts

```diff
@@ -90,14 +91,20 @@ function makeKerberosClient(authContext: AuthContext, callback: Callback<Kerbero
         Object.assign(initOptions, { user: username, password: password });
       }
 
-      initializeClient(
-        `${serviceName}${process.platform === 'win32' ? '/' : '@'}${host}`,
-        initOptions,
-        (err: string, client: KerberosClient): void => {
-          if (err) return callback(new MongoDriverError(err));
-          callback(undefined, client);
+      let spn: string;
+      if (process.platform === 'win32') {
+        spn = `${serviceName}/${host}`;
+        if ('SERVICE_REALM' in mechanismProperties) {
+          spn = `${spn}@${mechanismProperties.SERVICE_REALM}`;
         }
-      );
+      } else {
+        spn = `${serviceName}@${host}`;
+      }
```

Comment:
We want to keep the logic that uses slash instead of @ on windows, but move attaching the service realm to outside so it gets attached regardless of platform.

Reply:
The test: `validate that SERVICE_REALM and CANONICALIZE_HOST_NAME can be passed in`
in: `test/manual/kerberos.test.js:54`
is now going to fail after making this change, you can safely skip it, we have a ticket (NODE-3060) to track unskipping / removing later.

Reply:
I discussed that with @mcasimir, too... So we want to just attach that regardless of the platform _knowing_ it will not work on any *nix systems? I've no hard feelings on that - I'll add it :)

Reply:
Yea I'm just looking at python and mirroring their logic, we will revisit this when we have the right testing environment. You were able to confirm it works on windows so I'm happy with what we have here, thanks for this!